### DISCUSSION TITLE
Faster code for SelectIntegrationFeatures()

### DISCUSSION THREAD

**Discussion by danielcgingerich** [satijalab/seurat#4607] [Q&A] [answered]
I recently ran SelectIntegrationFeatures() on 24 atac datasets.  Each dataset has 276536 features.  SelectIntegrationFeatures() has been running on the university hpc for 16 hours now.  It seems the issue comes when ranking features by median across datasets.  So I thought I would try and make a faster code for this, in case anyone else is running into the same issue.  

How does this code look?  After all the datasets were loaded, it took about 60 seconds to run.

```
atac.list <- list()
for (i in 1:24){
  message(i)
  atac.list[[i]] <- readRDS(paste0(main.dir, rds.list[i]))
}


vf.list <- lapply(atac.list, VariableFeatures)

ranking <- c(1:nrow(atac.list[[1]]))
reorder <- rownames(atac.list[[1]])

for (i in 1:24){
  names(vf.list[[i]]) <- ranking
  vf.list[[i]] <- vf.list[[i]][match(reorder, vf.list[[i]])]
}

vf.ranks <- matrix(as.numeric(unlist(lapply(vf.list, names))), ncol = 24, byrow = F)

vf.medians <- apply(vf.ranks, MARGIN = 1, FUN = median)
names(vf.medians) <- reorder
vf.medians <- sort(vf.medians, decreasing = F)

integration.features <- names(vf.medians)[1:5000]
```

**Comment by timoast** [comment, marked as answer]
I would not recommend running `SelectIntegrationFeatures()` for scATAC-seq. Instead you should do the following:

1. Quantify the same peaks across all datasets to be integrated
2. Merge datasets and compute LSI
3. Check for batch effect
4. Find anchors between datasets using LSI projection
5. Integrate LSI embeddings

See https://satijalab.org/signac/articles/integrate_atac.html

In the LSI-projection workflow integration features are not needed, all features used to compute the LSI are used.

**Reply by fahmadimoughari** [reply]
The link does not work, can you please share a new link or let us know what to set as anchor.features in FindIntegrationAnchors function?
I have 6 samples to integrate and I am using the following code:
`integration.anchors <- FindIntegrationAnchors(
			object.list = sample.list,
			anchor.features = features,
			reduction = "rlsi",
			reference = 2:6,
			dims = 2:30
		);

integrated <- IntegrateEmbeddings(
		anchorset = integration.anchors,
		reductions = combined[["lsi"]],
		new.reduction.name = "integrated_lsi",
		dims.to.integrate = 1:30
	);`
If I don't pass anchor.features parameter the run will be killed even with 48 cores on cluster because of lack of storage.
